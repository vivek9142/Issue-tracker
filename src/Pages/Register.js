import React,{useEffect} from 'react';
import {Formik,Form} from 'formik';
import TextField from '../Components/TextField';
import * as Yup from 'yup';

import { useDispatch,useSelector } from 'react-redux';
import { registerUser } from '../Redux/Actions/UserActions';
import { userActions } from '../Redux/Reducer/userReducer';

const Register = (props) => {
    const  dispatch = useDispatch();

    useEffect(() => {
        return () => {
            dispatch(userActions.resetErrorStatus());
        }
    },[]);

    const regError = useSelector(state => state.user.error);

    const validate = Yup.object({
        firstName:Yup.string()
        .required('First Name is Required'),
        lastName:Yup.string()
        .required('Last Name is Required'),
        location:Yup.string()
        .required('Location is required'),
        email:Yup.string()
        .email('Email is invalid')
        .required('Email is Required'),
        mobNo:Yup.string()
        .min(10,'Mobile Number is invalid')
        .max(10,'Mobile Number is invalid')
        .required('Mobile Number is required'),
        password:Yup.string()
        .min(6,'Password must be atleast 6 charaters long')
        .required('Password is Required'),
        passwordConf:Yup.string()
        .oneOf([Yup.ref('password'),null],'Password should match')
        .required('Confirm Password is Required')
    })

    const registerHandler = (user) => {
        user.passwordConf = undefined;
        dispatch(registerUser(user))
        
    }
    return (
        <>
        <h1>Register Page</h1>
        <span>{regError}</span>
        <Formik initialValues={{
            firstName:''
            ,lastName:''
            ,location:''
            ,email:''
            ,mobNo:''
            ,password:''
            ,passwordConf:''
        }} 
        validationSchema={validate} 
        onSubmit={val => registerHandler(val)}>

            {formik => (
                <Form>
                    <TextField type='text' name='firstName' label='First Name'/>
                    <TextField type='text' name='lastName' label='Last Name'/>
                    <TextField type='text' name='location' label='Location'/>
                    <TextField type='email' name='email' label='Email'/>
                    <TextField type='number' name='mobNo' label='Mobile Number'/>
                    <TextField type='password' name='password' label='Password'/>
                    <TextField type='password' name='passwordConf' label='Password Confirm'/>
                    <input type='submit'/>
                </Form>
            )}
        </Formik>
        </>
    );
}

export default Register;