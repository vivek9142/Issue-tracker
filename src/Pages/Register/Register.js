import React,{useEffect} from 'react';
import {Formik,Form} from 'formik';
import TextField from '../../Components/TextField/TextField';
import * as Yup from 'yup';

import { useDispatch,useSelector } from 'react-redux';
import { registerUser } from '../../Redux/Actions/UserActions';
import { userActions } from '../../Redux/Reducer/userReducer';

const Register = (props) => {
    const  dispatch = useDispatch();

    useEffect(() => {
        return () => {
            dispatch(userActions.resetErrorStatus());
        }
    },[]);

    const regError = useSelector(state => state.user.error);

    const validate = Yup.object({
        firstName:Yup.string()
        .required('First Name is Required'),
        lastName:Yup.string()
        .required('Last Name is Required'),
        location:Yup.string()
        .required('Location is required'),
        email:Yup.string()
        .email('Email is invalid')
        .required('Email is Required'),
        mobNo:Yup.string()
        .min(10,'Mobile Number is invalid')
        .max(10,'Mobile Number is invalid')
        .required('Mobile Number is required'),
        password:Yup.string()
        .min(6,'Password must be atleast 6 charaters long')
        .required('Password is Required'),
        passwordConf:Yup.string()
        .oneOf([Yup.ref('password'),null],'Password should match')
        .required('Confirm Password is Required')
    })

    const registerHandler = (user) => {
        user.passwordConf = undefined;
        dispatch(registerUser(user))
        
    }
    return (
        <>
        <div className="container user-section mt-5 mb-3">
            <h3 className="text-center">
                Register 
            </h3>
            <hr/>
        </div>
        <div className="container mb-5">
            <h5 className="fs-6 text-center fw-normal ">{regError}
            </h5>
        </div>
        <Formik initialValues={{
            firstName:''
            ,lastName:''
            ,location:''
            ,email:''
            ,mobNo:''
            ,password:''
            ,passwordConf:''
        }} 
        validationSchema={validate} 
        onSubmit={val => registerHandler(val)}>

            {formik => (
                <>
                <div className="container user-section--auth-container">
                    <Form>
                    <div className="row g-3">
                            <div className="col-md-6">
                                <TextField type='text' name='firstName' label='First Name'/>
                            </div>
                            <div className="col-md-6">
                                <TextField type='text' name='lastName' label='Last Name'/>
                            </div>
                            <div className="col-md-6">
                                <TextField type='text' name='location' label='Location'/>
                            </div>
                            <div className="col-md-3">
                                <TextField type='email' name='email' label='Email'/>
                            </div>
                            <div className="col-md-3">
                                <TextField type='number' name='mobNo' label='Mobile Number'/>
                            </div>
                            <div className="col-md-6">
                                <TextField type='password' name='password' label='Password'/>
                            </div>
                            <div className="col-md-6">
                                <TextField type='password' name='passwordConf' label='Password Confirm'/>
                            </div>

                            <div className="col-12 " style={{'marginBottom':'5rem'}}>
                                <button type="submit" className="btn btn-primary">Register</button>
                            </div>
                    </div>
                    </Form>
                </div>
                </>
            )}
        </Formik>
        
        </>
    );
}

export default Register;