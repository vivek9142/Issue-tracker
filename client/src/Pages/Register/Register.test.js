import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Register from './Register';
import {Provider} from 'react-redux';
import store from '../../Redux/Store';
import { MemoryRouter } from 'react-router-dom';

describe('Register page Page Test Suite', () => {
    beforeEach(()=>{
        
    render(
        <MemoryRouter>   
        <Provider store={store}><Register/></Provider>
        </MemoryRouter> 
        );
    });
    test('Register page Snapshot',()=>{
        const tree = renderer.create(<MemoryRouter>   
            <Provider store={store}><Register/></Provider>
            </MemoryRouter> );
        expect(tree).toMatchSnapshot();
      });
      
    test('renders main Heading of Register page Page', () => {
        const element = screen.getByRole('button', { role:3 });;
        expect(element).toBeInTheDocument();
    });

    test('Register page should have inputs', () => {
        expect(screen.getAllByRole('textbox')).toHaveLength(4);
        expect(screen.getAllByRole('spinbutton')).toHaveLength(1);
    });

    test('Register page should have a submit button', () => {
        const submitButton = screen.getByRole('button', { name: 'Register' });
        expect(submitButton).toBeDefined()
    });

  
})