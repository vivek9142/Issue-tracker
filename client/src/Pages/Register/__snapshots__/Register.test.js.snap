// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page Page Test Suite Register page Snapshot 1`] = `
Array [
  <div
    className="container user-section mt-5 mb-3"
  >
    <h3
      className="text-center"
    >
      Register
    </h3>
    <hr />
  </div>,
  <div
    className="container mb-5"
  >
    <h5
      className="fs-6 text-center fw-normal "
    >
      
    </h5>
  </div>,
  <div
    className="container user-section--auth-container"
  >
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="row g-3"
      >
        <div
          className="col-md-6"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="First Name"
            >
              First Name
            </label>
            <input
              className="form-control"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your First Name"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Last Name"
            >
              Last Name
            </label>
            <input
              className="form-control"
              name="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Last Name"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Location"
            >
              Location
            </label>
            <input
              className="form-control"
              name="location"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Location"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Email"
            >
              Email
            </label>
            <input
              className="form-control"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Email"
              type="email"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-3"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Mobile Number"
            >
              Mobile Number
            </label>
            <input
              className="form-control"
              name="mobNo"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Mobile Number"
              type="number"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Password"
            >
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Password"
              type="password"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            className=""
          >
            <label
              className="form-label"
              htmlFor="Password Confirm"
            >
              Password Confirm
            </label>
            <input
              className="form-control"
              name="passwordConf"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter your Password Confirm"
              type="password"
              value=""
            />
            <div
              style={
                Object {
                  "height": "1rem",
                }
              }
            />
          </div>
        </div>
        <div
          className="col-12 "
          style={
            Object {
              "marginBottom": "5rem",
            }
          }
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Register
          </button>
        </div>
      </div>
    </form>
  </div>,
]
`;
